layers/core_validation.cpp

```diff
@@ -1577,6 +1577,39 @@ bool CoreChecks::ValidateCmdBufDrawState(const CMD_BUFFER_STATE *cb_node, CMD_TY
 
     bool result = false;
 
+    if (pipe->descriptor_buffer_mode) {
+        if (!last_bound.per_set.empty()) {
+            bool non_push = false;
+            for (const auto &ds : last_bound.per_set) {
+                non_push = ds.bound_descriptor_set && !ds.bound_descriptor_set->IsPushDescriptor();
+
+                if (non_push) break;
+            }
+
+            if (non_push)
+                result |= LogError(pipe->pipeline(), vuid.descriptor_buffer_bit_not_set,
+                                   "If the descriptors used by the VkPipeline bound to the pipeline bind point were specified via "
+                                   "vkCmdBindDescriptorSets, the bound VkPipeline must have been created without "
+                                   "VK_PIPELINE_CREATE_DESCRIPTOR_BUFFER_BIT_EXT");
+        }
+    } else {
+        if (cb_node->descriptor_buffer_bindings.size() > 0)
+            result |= LogError(pipe->pipeline(), vuid.descriptor_buffer_set_offset_missing,
+                               "If the descriptors used by the VkPipeline bound to the pipeline bind point were specified via "
+                               "vkCmdSetDescriptorBufferOffsetsEXT, the bound VkPipeline must have been created with "
+                               "VK_PIPELINE_CREATE_DESCRIPTOR_BUFFER_BIT_EXT");
+    }
+
+    if (pipe->descriptor_buffer_mode) {
+        if (!cb_node->descriptor_buffer_bindings.empty()) {
+            // TODO-08116 Descriptors in bound descriptor buffers, specified via vkCmdSetDescriptorBufferOffsetsEXT, must be valid
```

Comment:
Please tie new TODO to issues (new system we are trying to enforce)

```suggestion
            // TODO Issue 4832 - VUID 08116 Descriptors in bound descriptor buffers, specified via vkCmdSetDescriptorBufferOffsetsEXT, must be valid
```